Incorporated ALE specific fields and general refactoring

pvphaneuf opened the pull request.

zakandrewking commented:
This is looking good. 

A couple questions:

1. I understand the goal of having a more structured `temperature` field. Should we also have an `environment` field for other environmental parameters? Will there be any?

2. It looks like we dropped the `biological replicates` field but kept `technical replicates`. We should probably keep both, or discuss dropping both if they are not useful.

3. Adam asked me to add the following to the interface. I haven't had a change, so if you find time to stick these in too, while you're working on it, that would be a huge help! (i'm happy to accept the PR either way :)

- Antibiotic Concentration (ug/mL)
- MIC Value
- Sample Processing Date  --> would have a calender
- Sample Time --> This would be used to say 2 hour from t0 or
something like that. A lot of times, we will take different samples
throughout.  I think from t=0 is the most common refernce.

zakandrewking commented:
Adam suggested the following changes too:

- 'Library Prep Kit Cycles' should become 'Machine Kit Cycles'
- add 'Library Prep Kit Type' as a manufacturer can have more than one kit type, even for a given experiment type

zakandrewking commented:
We are in the process of creating a procedure for updating metadata, historical and future

pvphaneuf commented:
We've recently implement some additions to this form that will allow for displaying different metadata workflows/fields depending on whether the user indicates their intent to generate a generic metadata file or ALE metadata file. This should therefore enable us to host the metadata form for the SBRG location for all metadata generation. We've added back all of the original fields that we had removed due to their vagueness in the context of ALE metadata; the metadata context switch allows us to hide these when ALE is specified. Take a look; it's currently live at https://pvphaneuf.github.io/experiment_metadata/. Let Denny and I know if there are more professional choices for controls and transitioning between "Generic" and "ALE".

A while ago, I had removed the _instructions_ at the top of the page. Are they still relavant for the generic metadata workflow?

Could you clarify some points from your previous comments?

- What is the "MIC" in "MIC Value"?
- What is the "Processing Date"? Is it when the sample was sequenced?

zakandrewking commented on index.html:
```diff
@@ -31,62 +36,28 @@
       <h1>
         Experimental Metadata
-        <small>[<a href="http://github.com/SBRG/experiment_metadata">GitHub</a>]</small>
+        <small>[<a href="http://github.com/SBRG/">GitHub</a>]</small>
```
why change this link?

pvphaneuf replied:
Changed back in the branch that this pull request was initiated from. I'm unsure if this pull request will in fact incorporate those new changes; let me know if it won't I can make a new pull request for all of the fixes I'm putting in according to this code review.

zakandrewking commented on index.html:
```diff
@@ -31,62 +36,28 @@
+        <small>[<a href="http://github.com/SBRG/">GitHub</a>]</small>
       </h1>
 
-      <p class="lead">
```
Let's keep these (lead & instructions) for the default mode. you can hide them in ALE mode

pvphaneuf replied:
Added the lead back in a recent push to my pull request branch.

pvphaneuf replied:
Added these instruction to be presented when toggled to default mode.

zakandrewking commented on index.html:
```diff
@@ -95,35 +66,37 @@
-          <div class="panel panel-default">
-            <div class="panel-heading">Folder name</div>
-            <div class="panel-body match-height">
-              <input id="folder-name" class="form-control" onclick="select();" readonly></input>
```
I would like to have this around, too, for the Default mode.

pvphaneuf replied:
Added these instruction to be presented when toggled to default mode.

pvphaneuf replied:
Added this functionality back into branch with the origin of the merge.

zakandrewking commented on main.js:
```diff
@@ -45,16 +45,23 @@ var data = [
                   '<a href="http://www.hgvs.org/mutnomen/recs.html" target="_blank" tabindex="-1">here</a>.'),
     required: true,
     example: 'e.g. Keio-crp, 76A>T, D111E, ΔF508, BOP8900(ΔadhE)' },
+  { label: 'Species',
```
this is redundant with ncbi taxonomy id

pvphaneuf replied:
Removed

zakandrewking commented on main.js:
```diff
@@ -155,117 +179,245 @@ var data = [
 
   if (_.isUndefined(input_only))
-    input_only = false
+    input_only = false;
```
semicolons are missing because i generally follow this style: http://standardjs.com

I'm sure I was inconsistent, so it might be good to choose a new JS style guide that we both like and mention it in the README (or CONTRIBUTING.md). Here's another one with semicolons: https://github.com/airbnb/javascript

pvphaneuf replied:
I'm down with http://standardjs.com

pvphaneuf replied:
Pushed to merge branch removal of all semicolons.

pvphaneuf commented:
OK; all requests should have been addressed. Check it out and LMK if there's anything else; if no other issues to address, we should merge so the upcoming projects can use the office SBRG form.
